### PR TITLE
Submit draft pick

### LINKED ISSUE #266: Owners can submit InSeasonDraftPicks
Would like to let owners submit their draft picks through the website.

### PR DESCRIPTION
**Pull request by axelclark**
Owners can submit draft picks from the Championship page for each sport
Closes #266

### CODE REVIEW

File: lib/ex338/in_season_draft_pick/store.ex [hunk 1]
```diff
@@ -0,0 +1,27 @@
+defmodule Ex338.InSeasonDraftPick.Store do
```
Comment (credoci) [inline lib/ex338/in_season_draft_pick/store.ex]: Modules should have a @moduledoc tag.

File: web/models/in_season_draft_pick.ex [hunk 2]
```diff
@@ -21,14 +22,46 @@ defmodule Ex338.InSeasonDraftPick do
     |> validate_required([:position, :draft_pick_asset_id, :championship_id])
   end
 
+  def owner_changeset(struct, params \\ %{}) do
```
Comment (credoci) [inline web/models/in_season_draft_pick.ex]: Parameter `struct` has same name as the `Kernel.struct` function.

File: web/models/abilities.ex [hunk 3]
```diff
@@ -24,4 +24,9 @@ defimpl Canada.Can, for: Ex338.User do
     waiver.fantasy_team.owners
     |> Enum.any?(&(&1.user_id == user_id))
   end
+
+  defp owner?(user_id, %InSeasonDraftPick{} = draft_pick) do
+    draft_pick.draft_pick_asset.fantasy_team.owners
+    |> Enum.any?(&(&1.user_id == user_id))
```
Comment (axelclark) [inline web/models/abilities.ex]: Use a function call when a pipeline is only one function long

File: test/models/fantasy_player_repo_test.exs [hunk 4]
```diff
@@ -107,6 +107,84 @@ defmodule Ex338.FantasyPlayerRepoTest do
...
+      _pick_player =
+        insert(:fantasy_player, player_name: "B", draft_pick: true,
+          sports_league: sport)
+      _other_sport_player =
+        insert(:fantasy_player, player_name: "A", draft_pick: false,
+          sports_league: other_sport)
+
+      result =
+        FantasyPlayer.avail_players_for_champ(FantasyPlayer, league.id, sport.id)
+        |> Repo.all
```
Comment (axelclark) [inline test/models/fantasy_player_repo_test.exs]: Use a function call when a pipeline is only one function long

File: lib/ex338/championship/store.ex [hunk 5]
```diff
@@ -71,4 +72,12 @@ defmodule Ex338.Championship.Store do
   defp add_rank(slot, acc) do
      {Map.put(slot, :rank, "-"), acc + 1}
   end
+
+  def update_next_in_season_pick(championship) do
+    picks =
+      championship.in_season_draft_picks
+      |> InSeasonDraftPick.update_next_pick
```
Comment (axelclark) [inline lib/ex338/championship/store.ex]: Use a function call when a pipeline is only one function long

File: test/models/in_season_draft_pick_test.exs [hunk 6]
```diff
@@ -41,8 +63,69 @@ defmodule Ex338.InSeasonDraftPickTest do
...
+      pick_owner  = insert(:owner, fantasy_team: team)
+      pick   = insert(:fantasy_player, draft_pick: true)
+      player = insert(:fantasy_player, draft_pick: false)
+      championship = insert(:championship)
+      pick_asset =
+        insert(:roster_position, fantasy_team: team, fantasy_player: pick)
+      insert(:in_season_draft_pick, drafted_player: player,
+        draft_pick_asset: pick_asset, championship: championship)
+
+
```
Comment (axelclark) [inline test/models/in_season_draft_pick_test.exs]: There should be no more than 1 consecutive blank lines.

File: web/views/view_helpers.ex [hunk 7]
```diff
@@ -11,7 +11,12 @@ defmodule Ex338.ViewHelpers do
     |> Enum.any?(&(&1.user_id == id))
   end
 
-  def owner?(current_user, draft_pick) do
+  def owner?(current_user, %InSeasonDraftPick{} = draft_pick) do
+    draft_pick.draft_pick_asset.fantasy_team.owners
+    |> Enum.any?(&(&1.user_id == current_user.id))
```
Comment (axelclark) [inline web/views/view_helpers.ex]: Use a function call when a pipeline is only one function long

**Comment by axelclark** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 5 possible new issues (including those that may have been commented here).

You can see more details about this review at https://ebertapp.io/github/axelclark/ex338/pulls/268.

File: lib/ex338/in_season_draft_pick/store.ex (hunk 1 above)
Comment (axelclark) [inline lib/ex338/in_season_draft_pick/store.ex]: k

File: web/models/in_season_draft_pick.ex (hunk 2 above)
Comment (axelclark) [inline web/models/in_season_draft_pick.ex]: k

File: web/models/abilities.ex (hunk 3 above)
Comment (axelclark) [inline web/models/abilities.ex]: k

File: test/models/fantasy_player_repo_test.exs (hunk 4 above)
Comment (axelclark) [inline test/models/fantasy_player_repo_test.exs]: k

File: lib/ex338/championship/store.ex (hunk 5 above)
Comment (axelclark) [inline lib/ex338/championship/store.ex]: k

File: test/models/in_season_draft_pick_test.exs (hunk 6 above)
Comment (axelclark) [inline test/models/in_season_draft_pick_test.exs]: k

**Comment by axelclark** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 1 possible new issue (including those that may have been commented here).
* 3 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/axelclark/ex338/pulls/268.

**Pull request merged**